storage/client.go

```diff
@@ -48,6 +49,10 @@ const (
 	userAgentHeader = "User-Agent"
 )
 
+var (
+	validStorageAccount = regexp.MustCompile("^[0-9a-z]{3,24}$")
```

Comment:
Without adding anything, Azure would validate the account name by returning a failure. :) I think the idea is just to save people a request. Though, I do agree with the principal of not encoding unnecessary logic.